### PR TITLE
RBAC: Make RBAC action names more consistent

### PR DESCRIPTION
**Pull request by IevaVasiljeva**
**What this PR does / why we need it**:
cleans up RBAC action names to be more consistent. It's way easier to do this before the GA release, so it'd be nice to get these changes in before them if we want them.

**Special notes for your reviewer**:
We can also go forward only with a subset of these changes if you prefer.

Action renames **will break** custom roles that use previous action names. We could add migrations to avoid that, but I think it's fine to do this while we're still in GA.

Enterprise PR: https://github.com/grafana/grafana-enterprise/pull/3372

# Release notice breaking change

The following RBAC action renames have been carried out:

* `users.authtoken:update` -> `users.authtoken:write`;
* `users.password:update` -> `users.password:write`;
* `users.permissions:update` -> `users.permissions:write`;
* `users.quotas:update` -> `users.quotas:write`;
* `org.users.role:update` -> `org.users:write`;
* `alert.instances:update` -> `alert.instances:write`;
* `alert.rules:update` -> `alert.rules:write`;
* `users.authtoken:list` -> `users.authtoken:read`;
* `users.quotas:list` -> `users.quotas:read`;
* `users.teams:read` -> replaced by `users.read` + `teams:read`

We've added a migration from the old action names to the new names and have updated our documentation. But you will have to update any scripts and provisioning files that are using the old action names.

### CODE REVIEW

**Review by chri2547 — APPROVED** [review]

File: pkg/services/serviceaccounts/database/database.go [hunk 1]
```diff
@@ -209,25 +209,7 @@ func (s *ServiceAccountsStoreImpl) RetrieveServiceAccount(ctx context.Context, o
...
-	}
-	teams := make([]string, len(getTeamQuery.Result))
-
-	for i := range getTeamQuery.Result {
-		teams[i] = getTeamQuery.Result[i].Name
-	}
-
-	serviceAccount.Teams = teams
-
-	return serviceAccount, nil
```
Comment (IevaVasiljeva) [inline pkg/services/serviceaccounts/database/database.go:230]: Service accounts currently can't be added to teams, so this functionality is safe to remove.

File: pkg/api/api.go [hunk 2]
```diff
@@ -179,7 +179,7 @@ func (hs *HTTPServer) registerRoutes() {
 			usersRoute.Get("/", authorize(reqGrafanaAdmin, ac.EvalPermission(ac.ActionUsersRead)), routing.Wrap(hs.searchUsersService.SearchUsers))
 			usersRoute.Get("/search", authorize(reqGrafanaAdmin, ac.EvalPermission(ac.ActionUsersRead)), routing.Wrap(hs.searchUsersService.SearchUsersWithPaging))
 			usersRoute.Get("/:id", authorize(reqGrafanaAdmin, ac.EvalPermission(ac.ActionUsersRead, userIDScope)), routing.Wrap(hs.GetUserByID))
-			usersRoute.Get("/:id/teams", authorize(reqGrafanaAdmin, ac.EvalPermission(ac.ActionUsersTeamRead, userIDScope)), routing.Wrap(hs.GetUserTeams))
+			usersRoute.Get("/:id/teams", authorize(reqGrafanaAdmin, ac.EvalPermission(ac.ActionUsersRead, userIDScope)), routing.Wrap(hs.GetUserTeams))
```
Comment (IevaVasiljeva) [inline pkg/api/api.go:182]: `users.teams:read` is removed in favour of `users:read` + `teams:read`. The check for `teams:read` permission is carried out using RBAC filter here: https://github.com/grafana/grafana/pull/49730/files#diff-9ff4b79076656e607d76f40213e6a3cff3e4bbf1ec0deda50dee277b25ebaae3R320-R329

File: pkg/services/sqlstore/team.go [hunk 3]
```diff
@@ -310,12 +310,23 @@ func (ss *SQLStore) GetTeamsByUser(ctx context.Context, query *models.GetTeamsBy
...
+		if !ac.IsDisabled(ss.Cfg) {
+			acFilter, err := ac.Filter(query.SignedInUser, "team.id", "teams:id:", ac.ActionTeamsRead)
+			if err != nil {
+				return err
+			}
+			sql.WriteString(` and` + acFilter.Where)
+			params = append(params, acFilter.Args...)
+		}
+
+		err := sess.SQL(sql.String(), params...).Find(&query.Result)
```
Comment (IevaVasiljeva) [inline pkg/services/sqlstore/team.go:329]: users.teams:read is removed in favour of users:read + teams:read. The check for `users:read` permission is carried out here: https://github.com/grafana/grafana/pull/49730/files#diff-67a6e36014be865053c43d38b2091bd2d3f17a91b117d97c0f5244cd7d71127dR182

File: pkg/services/accesscontrol/models.go [hunk 4]
```diff
@@ -285,32 +285,31 @@ const (
...
 	// Users actions
-	ActionUsersRead     = "users:read"
-	ActionUsersWrite    = "users:write"
-	ActionUsersTeamRead = "users.teams:read"
+	ActionUsersRead  = "users:read"
+	ActionUsersWrite = "users:write"
 	// We can ignore gosec G101 since this does not contain any credentials.
 	// nolint:gosec
-	ActionUsersAuthTokenList = "users.authtoken:list"
+	ActionUsersAuthTokenList = "users.authtoken:read"
```
Comment (kalleep) [inline pkg/services/accesscontrol/models.go:292]: nit: Change constant to ActionUsersAuthTokenRead

File: pkg/services/accesscontrol/models.go [hunk 5]
```diff
@@ -285,32 +285,31 @@ const (
...
 	// We can ignore gosec G101 since this does not contain any credentials.
 	// nolint:gosec
-	ActionUsersAuthTokenUpdate = "users.authtoken:update"
+	ActionUsersAuthTokenUpdate = "users.authtoken:write"
```
Comment (kalleep) [inline pkg/services/accesscontrol/models.go:295]: nit: Change constant to ActionUsersAuthTokenWrite

File: pkg/services/accesscontrol/models.go [hunk 6]
```diff
@@ -285,32 +285,31 @@ const (
...
 	// We can ignore gosec G101 since this does not contain any credentials.
 	// nolint:gosec
-	ActionUsersPasswordUpdate    = "users.password:update"
+	ActionUsersPasswordUpdate    = "users.password:write"
```
Comment (kalleep) [inline pkg/services/accesscontrol/models.go:298]: nit: Change constant to ActionUsersPasswordWrite

File: pkg/services/accesscontrol/models.go [hunk 7]
```diff
@@ -285,32 +285,31 @@ const (
...
 	ActionUsersDelete            = "users:delete"
 	ActionUsersCreate            = "users:create"
 	ActionUsersEnable            = "users:enable"
 	ActionUsersDisable           = "users:disable"
-	ActionUsersPermissionsUpdate = "users.permissions:update"
+	ActionUsersPermissionsUpdate = "users.permissions:write"
```
Comment (kalleep) [inline pkg/services/accesscontrol/models.go:303]: nit: Change constant to ActionUsersPermissionsWrite

File: pkg/services/accesscontrol/models.go [hunk 8]
```diff
@@ -285,32 +285,31 @@ const (
...
 	ActionUsersLogout            = "users:logout"
-	ActionUsersQuotasList        = "users.quotas:list"
-	ActionUsersQuotasUpdate      = "users.quotas:update"
+	ActionUsersQuotasList        = "users.quotas:read"
```
Comment (kalleep) [inline pkg/services/accesscontrol/models.go:305]: nit: Change constant to ActionUsersQuotasRead

File: pkg/services/accesscontrol/models.go [hunk 9]
```diff
@@ -285,32 +285,31 @@ const (
...
+	ActionUsersQuotasUpdate      = "users.quotas:write"
```
Comment (kalleep) [inline pkg/services/accesscontrol/models.go:306]: nit: Change constant to ActionUsersQuotasWrite

File: pkg/services/accesscontrol/models.go [hunk 10]
```diff
@@ -363,12 +362,12 @@ const (
 	// Alerting rules actions
 	ActionAlertingRuleCreate = "alert.rules:create"
 	ActionAlertingRuleRead   = "alert.rules:read"
-	ActionAlertingRuleUpdate = "alert.rules:update"
+	ActionAlertingRuleUpdate = "alert.rules:write"
```
Comment (kalleep) [inline pkg/services/accesscontrol/models.go:365]: nit: Change constant to ActionAlertingRuleWrite

File: pkg/services/accesscontrol/models.go [hunk 11]
```diff
@@ -363,12 +362,12 @@ const (
...
 	ActionAlertingRuleDelete = "alert.rules:delete"
 
 	// Alerting instances (+silences) actions
 	ActionAlertingInstanceCreate = "alert.instances:create"
-	ActionAlertingInstanceUpdate = "alert.instances:update"
+	ActionAlertingInstanceUpdate = "alert.instances:write"
```
Comment (kalleep) [inline pkg/services/accesscontrol/models.go:370]: nit: Change constant to ActionAlertingInstanceWrite

File: pkg/services/licensing/accesscontrol.go [hunk 12]
```diff
@@ -4,7 +4,7 @@ import "github.com/grafana/grafana/pkg/services/accesscontrol"
 
 const (
 	ActionRead        = "licensing:read"
-	ActionUpdate      = "licensing:update"
+	ActionUpdate      = "licensing:write"
```
Comment (kalleep) [inline pkg/services/licensing/accesscontrol.go:7]: nit: Change constant to ActionWrite

File: public/app/types/accessControl.ts [hunk 13]
```diff
@@ -10,18 +10,17 @@ export type UserPermission = Record<string, boolean>;
 export enum AccessControlAction {
   UsersRead = 'users:read',
   UsersWrite = 'users:write',
-  UsersTeamRead = 'users.teams:read',
-  UsersAuthTokenList = 'users.authtoken:list',
-  UsersAuthTokenUpdate = 'users.authtoken:update',
-  UsersPasswordUpdate = 'users.password:update',
+  UsersAuthTokenList = 'users.authtoken:read',
```
Comment (kalleep) [inline public/app/types/accessControl.ts:13]: nit: Change names to reflect the values for all below

**Review by kalleep — COMMENTED** [review]
Looks good. Left a comments. Need to test this before i can approve :)

File: pkg/services/accesscontrol/models.go (hunk 4 above)
Comment (IevaVasiljeva) [reply pkg/services/accesscontrol/models.go:292]: Do you mind if I change the constant names in a separate PR? This one is pretty big already, and I don't want it to grow even huger. I promise I'll actually do it 😄

Comment (kalleep) [reply pkg/services/accesscontrol/models.go:292]: Yeah i think that would be better :)

**Review by kalleep — APPROVED** [review]

**Comment by gamab** [comment]
Should this be back-ported to v8.5.x?

**Pull request merged**